deps/Makefile

```diff
@@ -1957,26 +2002,28 @@ LIBGIT2_OPTS += -DCMAKE_FIND_ROOT_PATH=/usr/$(XC_HOST) -DCMAKE_FIND_ROOT_PATH_MO
 endif
 endif
 
-$(LIBGIT2_SRC_DIR)/build/Makefile: $(LIBGIT2_SRC_DIR)/CMakeLists.txt
-	-cd $(LIBGIT2_SRC_DIR) && patch -p1 -f < ../libgit2_oldcmake.patch
-	-cd $(LIBGIT2_SRC_DIR) && patch -p1 -f < ../libgit2_oldssl.patch
-	mkdir -p $(LIBGIT2_SRC_DIR)/build
-	cd $(LIBGIT2_SRC_DIR)/build/ && \
-	$(CMAKE) .. $(LIBGIT2_OPTS)
+$(SRCDIR)/srccache/$(LIBGIT2_SRC_DIR)/patched: $(SRCDIR)/srccache/$(LIBGIT2_SRC_DIR)/CMakeLists.txt
+	cd $(dir $<) && patch -p1 -f < $(SRCDIR)/libgit2_oldcmake.patch
+	cd $(dir $<) && patch -p1 -f < $(SRCDIR)/libgit2_oldssl.patch
+	echo 1 > $@
+$(BUILDDIR)/$(LIBGIT2_SRC_DIR)/Makefile: $(SRCDIR)/srccache/$(LIBGIT2_SRC_DIR)/patched
+	mkdir -p $(dir $@)
+	cd $(dir $@) && \
+	$(CMAKE) $(dir $<) $(LIBGIT2_OPTS)
 	touch -c $@
-$(LIBGIT2_OBJ_SOURCE): $(LIBGIT2_SRC_DIR)/build/Makefile
-	$(MAKE) -C $(LIBGIT2_SRC_DIR)/build
+$(LIBGIT2_OBJ_SOURCE): $(BUILDDIR)/$(LIBGIT2_SRC_DIR)/Makefile
+	$(CMAKE) --build $(dir $<)
```

Comment:
That link doesn't actually point anywhere for me. Using this form can help a bit if you're using different generators like ninja, but you know what would help with that even more? Using cmake for as much of our build as possible. gmake doesn't know that `$(CMAKE) --build` will usually be a recursive make call.